spec.html

```diff
@@ -45766,19 +45770,11 @@ <h1>
           1. Let _asyncContext_ be the running execution context.
           1. Let _promise_ be ? PromiseResolve(%Promise%, _value_).
           1. Let _fulfilledClosure_ be a new Abstract Closure with parameters (_value_) that captures _asyncContext_ and performs the following steps when called:
-            1. Let _prevContext_ be the running execution context.
-            1. Suspend _prevContext_.
-            1. Push _asyncContext_ onto the execution context stack; _asyncContext_ is now the running execution context.
-            1. <emu-meta effects="user-code">Resume the suspended evaluation of _asyncContext_</emu-meta> using NormalCompletion(_value_) as the result of the operation that suspended it.
-            1. Assert: When we reach this step, _asyncContext_ has already been removed from the execution context stack and _prevContext_ is the currently running execution context.
+            1. Perform Completion(RunSuspendedContext(_asyncContext_, NormalCompletion(_value_))).
```

Comment:
@jmdyck @bakkot Does ecmarkup complain if you remove these `Completion(...)`s? If so, we should allow them to be omitted when used with `Perform` since they're not actually entering the algorithm.

Reply:
Yes it does, e.g.: `error: RunSuspendedContext returns a Completion Record, but is not consumed as if it does (typecheck) at spec.html:46018:24`